sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/src/TimeSeriesInsightsClient.cs

```diff
@@ -239,5 +247,777 @@ public virtual Response<TimeSeriesModelSettings> UpdateModelSettingsDefaultTypeI
+                            .ListAsync(nextLink, _clientSessionId, cancellationToken)
+                            .ConfigureAwait(false);
+                        return Page.FromValues(getInstancesResponse.Value.Instances, getInstancesResponse.Value.ContinuationToken, getInstancesResponse.GetRawResponse());
+                    }
+                    catch (Exception ex)
+                    {
+                        scope.Failed(ex);
+                        throw;
+                    }
+                }
+
+                return PageableHelpers.CreateAsyncEnumerable(FirstPageFunc, NextPageFunc);
+            }
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Gets Time Series instances from the environment in pages.
+        /// </summary>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>The pageable list <see cref="Pageable{TimeSeriesInstance}"/> of Time Series instances belonging to the TSI environment and the http response.</returns>
+        /// <seealso cref="GetInstancesAsync(CancellationToken)">
+        /// See the asynchronous version of this method for examples.
+        /// </seealso>
+        public virtual Pageable<TimeSeriesInstance> GetInstances(CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(TimeSeriesInsightsClient)}.{nameof(GetInstances)}");
+            scope.Start();
+
+            try
+            {
+                Page<TimeSeriesInstance> FirstPageFunc(int? pageSizeHint)
+                {
+                    using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(TimeSeriesInsightsClient)}.{nameof(GetInstances)}");
+                    scope.Start();
+
+                    try
+                    {
+                        Response<GetInstancesPage> getInstancesResponse = _timeSeriesInstancesRestClient.List(null, _clientSessionId, cancellationToken);
+                        return Page.FromValues(getInstancesResponse.Value.Instances, getInstancesResponse.Value.ContinuationToken, getInstancesResponse.GetRawResponse());
+                    }
+                    catch (Exception ex)
+                    {
+                        scope.Failed(ex);
+                        throw;
+                    }
+                }
+
+                Page<TimeSeriesInstance> NextPageFunc(string nextLink, int? pageSizeHint)
+                {
+                    using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(TimeSeriesInsightsClient)}.{nameof(GetInstances)}");
+                    scope.Start();
+
+                    try
+                    {
+                        Response<GetInstancesPage> getInstancesResponse = _timeSeriesInstancesRestClient.List(nextLink, _clientSessionId, cancellationToken);
```

Comment:
For client session Id, need to think how to provide it back the user. I will take this as a todo next.